x-pack/plugins/security_solution/public/management/pages/policy/view/policy_settings_form/components/detect_prevent_protection_lavel.tsx

```diff
@@ -0,0 +1,188 @@
+                  <ProtectionRadio
+                    policy={policy}
+                    onChange={onChange}
+                    mode={mode}
+                    protection={protection}
+                    protectionMode={id}
+                    osList={osList}
+                    label={label}
+                    data-test-subj={getTestId(`${id}Radio`)}
+                  />
+                </EuiFlexItem>
+              );
+            })
+          ) : (
+            <>{currentProtectionLevelLabel}</>
+          )}
+        </EuiFlexGroup>
+      </div>
+    );
+  }
+);
+DetectPreventProtectionLevel.displayName = 'DetectPreventProtectionLevel';
+
+interface ProtectionRadioProps extends PolicyFormComponentCommonProps {
+  protection: PolicyProtection;
+  protectionMode: ProtectionModes;
+  osList: ImmutableArray<Partial<keyof UIPolicyConfig>>;
+  label: string;
+}
+
+const ProtectionRadio = React.memo(
+  ({
+    protection,
+    protectionMode,
+    osList,
+    label,
+    onChange,
+    policy,
+    mode,
+    'data-test-subj': dataTestSubj,
+  }: ProtectionRadioProps) => {
+    const radioButtonId = useGeneratedHtmlId();
+    const selected = policy.windows[protection].mode;
+    const isPlatinumPlus = useLicense().isPlatinumPlus();
+    const showEditableFormFields = mode === 'edit';
+
+    const handleRadioChange = useCallback(() => {
+      const newPayload = cloneDeep(policy);
+
+      for (const os of osList) {
+        if (os === 'windows') {
+          newPayload[os][protection].mode = protectionMode;
+        } else if (os === 'mac') {
+          newPayload[os][protection as MacPolicyProtection].mode = protectionMode;
+        } else if (os === 'linux') {
+          newPayload[os][protection as LinuxPolicyProtection].mode = protectionMode;
+        }
+        if (isPlatinumPlus) {
+          if (os === 'windows') {
+            if (protectionMode === ProtectionModes.prevent) {
+              newPayload[os].popup[protection].enabled = true;
+            } else {
+              newPayload[os].popup[protection].enabled = false;
+            }
+          } else if (os === 'mac') {
+            if (protectionMode === ProtectionModes.prevent) {
+              newPayload[os].popup[protection as MacPolicyProtection].enabled = true;
+            } else {
+              newPayload[os].popup[protection as MacPolicyProtection].enabled = false;
+            }
+          } else if (os === 'linux') {
+            if (protectionMode === ProtectionModes.prevent) {
+              newPayload[os].popup[protection as LinuxPolicyProtection].enabled = true;
+            } else {
+              newPayload[os].popup[protection as LinuxPolicyProtection].enabled = false;
+            }
+          }
+        }
+      }
+
+      onChange({ isValid: true, updatedPolicy: newPayload });
+    }, [isPlatinumPlus, onChange, osList, policy, protection, protectionMode]);
+
+    // FIXME:PT Why is this className being used below? can it be removed?
```

Comment:
I don't find any styles related to the classname either. Maybe okay to remove it already?